Lib/dbm/sqlite3.py

```diff
@@ -0,0 +1,117 @@
+import os
+import sqlite3
+from contextlib import suppress, closing
+from collections.abc import MutableMapping
+
+BUILD_TABLE = """
+  CREATE TABLE IF NOT EXISTS Dict (
+    key TEXT NOT NULL,
+    value BLOB NOT NULL,
+    PRIMARY KEY (key)
+  )
+"""
+GET_SIZE = "SELECT COUNT (key) FROM Dict"
+LOOKUP_KEY = "SELECT value FROM Dict WHERE key = ? LIMIT 1"
+STORE_KV = "REPLACE INTO Dict (key, value) VALUES (?, ?)"
+DELETE_KEY = "DELETE FROM Dict WHERE key = ?"
+ITER_KEYS = "SELECT key FROM Dict"
+
+
+class error(OSError):
+    pass
+
+
+def _normalize_uri_path(path):
```

Comment:
What if filename contains "%"?

Reply:
Let's find out; I implemented ~~all~~ most of SQLite's recommendations for URI normalisation: https://sqlite.org/uri.html

I see I did not address the backslash case; I'll fix that and add more tests.

Reply:
I rewrote this using pathlib and also added substitution for "%".

Reply:
Windows paths are ... irritating. I'll have to look at those later; for now, I marked the Windows tests as WIP.

Reply:
Pushed 08c38481c10d8cbfcc9ee02f2133a3285eed216b which just uses pathlib's `as_uri()` with some amendment. I guess it's going to break on Windows. Anyway, I'm calling it a day. Thanks for the input so far!